python/proton/_delivery.py

```diff
@@ -34,7 +34,7 @@
 
 
 class NamedInt(int):
-    values = {}  # type: Dict[int, str]
```

Comment:
Flake 8 looks inside # type: comments and complains if it is using something that wasn't declared; I'd leave the comment, type annotations are awesome and when Python 2.7 is dropped, they can be used more fully.

Reply:
The doc for mypy says that Dict is available in typing for 2.7 - not sure what's going on here, but I strongly think we should ignore or silence flake8 here.

Reply:
```python
>>> import typing
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named typing
```

Has to be installed with pip; it is part of stdlib only since 3.5 or maybe 3.6

Reply:
and the comment I put there is silencing flake8, thats the `# noqa` part of it

Reply:
So make typing a dependency in the tox test so that it gets installed before running the test

Reply:
That's not enough, it has to be imported. And that would make `typing` a runtime dependency of the client. I could have a conditional import, of course, but that is IMO at the same level of messiness as the current state.

And Python 2.7 is out-of-support at this point anyways, and Proton will drop it in one or two versions. I think that the comment is fine until we go for 3.6 minimum in Proton, which will be in few months, as I understand it.

Reply:
But when you run flake8 under tox you're not specifying which version of python at all, so I don't see how it's complaining about py2

Reply:
Also re type comments - I'm much more interested in typeshed (as in separate) function level type annotations - I think I put that comment there as a helpful piece of user level info to help me understand the code. 
I don't think having flake8 check this is very useful tbh, so just stop it checking type annotation for the moment.
If I want type checking I'm going to use mypy

Reply:
Whether flake8 runs under pyhotn2 or python3 probably plays a role here. In Python 3 code, you are supposed to have that `from typing import Dict`; if you forget, it's an error that flake8 can help you with.

The flake8 error this produced is

```
30: ./proton/_delivery.py:37:18: F821 undefined name 'Dict'
```

so exactly same result as if you used that name in your code. I don't see a way to tell flake8 not to behave like this, except with the # noqa comment. There is only a single type annotation like this. I strongly believe that what's in the PR is adequate until Proton moves to 3.6. Then we can simply do `from typing import` and we'll do type annotations the 3.x way.

Reply:
One reason we may want this flake8 behavior is that it quickly finds typos in type annotations. Mypy runs significantly slower.

Reply:
Just add the # noqa without the extra comment then - that maintains the informal comment nature. I care much less about this sort of annotation than function level which will be useful to people using the library in an IDE

Reply:
I wanted to mark the line for revisiting when we drop Python 2.7. Because then I can delete the # noqa and add from typing import Dict at the top of the file, and maybe rewrite the type annotation to PEP 526 syntax.